packages/webapp/components/layouts/MainFeedPage.tsx

```diff
@@ -25,11 +37,17 @@ export default function MainFeedPage({
   children,
 }: MainFeedPageProps): ReactElement {
   const router = useRouter();
+  const { user } = useContext(AuthContext);
+  const { flags } = useContext(FeaturesContext);
+  const shouldShowMyFeed = isFeaturedEnabled(Features.MyFeedOn, flags);
   const [feedName, setFeedName] = useState(getFeedName(router?.pathname));
   const [isSearchOn, setIsSearchOn] = useState(router?.pathname === '/search');
 
   useEffect(() => {
-    if (router?.pathname === '/search') {
+    const isMyFeed = router?.pathname === '/my-feed';
+    if ((isMyFeed && !shouldShowMyFeed) || (isMyFeed && !user)) {
+      router.replace('popular');
```

Comment:
Same comments as above. I would even turn this condition into a function that can be utilized in both places.
Also, make sure to redirect to the home page and not to popular.

Reply:
I agree, let me turn this into a function to have an easier decision with the conditions.